engine/polyglot-api/src/main/java/org/enso/polyglot/HostAccessFactory.java

```diff
@@ -0,0 +1,35 @@
+package org.enso.polyglot;
+
+import org.graalvm.polyglot.HostAccess;
+
+/** Utility class for creating HostAccess object. */
+public class HostAccessFactory {
+
+  // Copied from com.oracle.truffle.api.interop.NumberUtils
+  // since the latter is inaccessible
+  private static final long LONG_MAX_SAFE_DOUBLE = 9007199254740991L; // 2 ** 53 - 1
+
+  public HostAccess allWithTypeMapping() {
+    return HostAccess.newBuilder()
+        .allowPublicAccess(true)
+        .allowAllImplementations(true)
+        .allowAllClassImplementations(true)
+        .allowArrayAccess(true)
+        .allowListAccess(true)
+        .allowBufferAccess(true)
+        .allowIterableAccess(true)
+        .allowIteratorAccess(true)
+        .allowMapAccess(true)
+        .allowAccessInheritance(true)
+        .targetTypeMapping(
+            Long.class,
+            Double.class,
```

Comment:
Adding a `targetTypeMapping` was considered when doing the `BigInteger` support: #3865 - at the end I haven't used it as that only works with engine properly set up - e.g. it is only going to work when one calls `allWithTypeMapping` - it is not going to work in other tests or when one embeds Enso language into own application.

Reply:
Yes, I didn't add it to tests on purpose. We don't seem to use the same context definition in all of them anyway and I believe that's OK.
I looked at the sources of Truffle and it seems that `targetTypeMapping` is the only way to resolve the problem of this ticket without replicating some dispatch logic from Truffle in our backend (which would be horrible).

Reply:
True. Truffle doesn't support _"loosely conversion"_ which is what we want here.